third_party/terraform/resources/resource_dataflow_job.go

```diff
@@ -499,3 +507,33 @@ func resourceDataflowJobIsVirtualUpdate(d *schema.ResourceData) bool {
 
 	return false
 }
+
+func waitForDataflowJobToBeUpdated(d *schema.ResourceData, config *Config, replacementJobID string, timeout time.Duration) error {
+	return resource.Retry(timeout, func() *resource.RetryError {
+		project, err := getProject(d, config)
+		if err != nil {
+			return resource.NonRetryableError(err)
+		}
+
+		region, err := getRegion(d, config)
+		if err != nil {
+			return resource.NonRetryableError(err)
```

Comment:
@emilymye @c2thorn,

@spew brought up the following about this line:

> This seems like a place where we would want to retry and thus not return NonRetryableError? 
> Example: transient errors such as service 500s, TLS handshakes, etc, I believe will be returned by resourceDataflowJobGetJob(...) as that function is simply using the GCP APIs directly.

Thoughts?

Reply:
Sounds plausible, but haven't seen any such errors in practice. It doesn't hurt to retry if we know specifically which errors we want to retry for

Reply:
It has occurred for many resources for us in the past. Most of it was fixed by running things through the retry functions in retry_utils.go using the defaultRetryPredicates in error_retry_predicates.

Reply:
suggest using the default retry predicates

Reply:
Ok, I'll put out a patch for Magic Modules, then if it looks good to the Terraform team, I can bring that patch to KCC's copy of Terraform to ensure we're in sync.

Reply:
Sounds good @jcanseco!

Reply:
Also apologies, I just realized I commented on the "NonRetryableError" for getRegion(). I meant to do so for the one for resourceDataflowGetJob(). Might've been obvious but I thought I should clarify it.